### DISCUSSION TITLE
Lettuce connection pool metrics

### DISCUSSION THREAD

**Discussion by tvaingart** [redis/lettuce#2342] [Q&A]
Hi,
we're currently configured with lettuce for all of our redis pools in production.
we're have an assumption that our pool is not configured correctly, and not optimized for our workload - for example, one of the metrics that we're missing is the number of active threads in the pool.

configuring the pool size to be too low, might result an artificial bottlenecks in pool, as most of the tasks will be queued.

is there any kind of live metric that can be expose to query the pool indicators on runtime? (we did found some static metrics by JMX, and even a NumActive metric in JMX, but it seems to be static and set to 1 all time).
our pool configured with max of 1K threads.

Thanks
Tom

**Comment by mp911de** [comment]
There are no active threads in a Pool. Well, while you can enable background eviction, there would be one thread associated with evicting connections in the background. 

Generally, there aren't many reasons why you would like to use connection pooling. Using transactions or blocking commands is an indicator for pool usage. Other than that, if you use `ConnectionPoolSupport`, then you get hold of `GenericObjectPool` that you can hook up for metric integration.
